src/cc_mm.c

```diff
@@ -82,6 +82,22 @@ _cc_realloc(void *ptr, size_t size, const char *name, int line)
     return p;
 }
 
+void *
+_cc_realloc_test(void *ptr, size_t size, const char *name, int line)
```

Comment:
I'm not so sure this would be cleaner. The log messages and the return statement are the only duplicated lines, and using macros within the function body is messier than having them outside.